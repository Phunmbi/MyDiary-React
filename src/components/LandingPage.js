import React from 'react';

const LandingPage = () => {
  return(
    <div className="hero">
      <div className="title">
        <h1>MyDiary</h1>
      </div>

      <section>
        <div className="hero-marketing">
            <p>Let's help you make it easier to keep your diary. </p>
            <p>With the ability to edit and delete entries as you please.</p>
            <p>All wrapped in a lightweight app built for your convenience.</p>
        </div>
        <div className="hero-account">
            <h3>Sign in</h3>
            <p id="errorResponse"></p>
            <form>
                <input id="email" type="text" placeholder="Email" required />
                <input id="password" type="password" placeholder="Password" required />
                <input id="submit" type="submit" value="Sign in" />
            </form>
            <p>Don't have an account? then
                <a href="signup.html">Sign up</a>
            </p>
        </div>
      </section>
    </div>
  )
};

export default LandingPage;
