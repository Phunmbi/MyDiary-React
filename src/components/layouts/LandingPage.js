import React from 'react';
import Signup from '../auth/Signup';

const LandingPage = (props) => {

  const handleClick = () => {
    console.log(props);
  }

  return (
    <div className="hero">
      <div className="title">
        <h1 onClick={handleClick}>MyDiary</h1>
      </div>

      <section>
        <div className="hero-marketing">
          <p>Let's help you make it easier to keep your diary. </p>
          <p>With the ability to edit and delete entries as you please.</p>
          <p>All wrapped in a lightweight app built for your convenience.</p>
        </div>
          <Signup {...props} />
      </section>
    </div>
  );
};

export default LandingPage;
