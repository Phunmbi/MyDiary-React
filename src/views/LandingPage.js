import React from 'react';

import { Link } from 'react-router-dom';

const LandingPage = () => {
  return(
    <div className="hero">
      <h2>My Diary</h2>
      <p><Link to='/auth'>sign up</Link></p>
    </div>
  );
};

export default LandingPage;
