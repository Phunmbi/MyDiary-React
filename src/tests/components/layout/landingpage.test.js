import React from 'react';
import toJson from 'enzyme-to-json';
import LandingPage from '../../../components/layouts/LandingPage';

let wrapper;

beforeEach(() => {
  jest.useFakeTimers();
  wrapper = shallow(
    <LandingPage />,
  );
});

afterEach(() => wrapper.unmount());

describe('Authentication UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
