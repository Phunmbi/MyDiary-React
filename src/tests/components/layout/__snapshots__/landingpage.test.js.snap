// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication UI render features component should render as expected 1`] = `
<div
  className="hero"
>
  <div
    className="title"
  >
    <h1>
      MyDiary
    </h1>
  </div>
  <section>
    <div
      className="hero-marketing"
    >
      <p>
        Let's help you make it easier to keep your diary. 
      </p>
      <p>
        With the ability to edit and delete entries as you please.
      </p>
      <p>
        All wrapped in a lightweight app built for your convenience.
      </p>
    </div>
    <Connect(Signup) />
  </section>
</div>
`;
